tests/sources/fixtures/sharepoint_online/connector.json

```diff
@@ -75,14 +75,64 @@
         "value": "*",
         "order": 5,
         "ui_restrictions": []
+      },
+      "use_text_extraction_service": {
+        "depends_on": [],
+        "display": "toggle",
+        "tooltip": "Requires a separate deployment of the Elastic Text Extraction Service. Requires that pipeline settings disable text extraction.",
+        "default_value": null,
+        "label": "Use text extraction service",
+        "sensitive": false,
+        "type": "bool",
+        "required": true,
+        "options": [],
+        "validations": [],
+        "value": false,
+        "order": 6,
+        "ui_restrictions": []
+      },
+      "use_document_level_security": {
+        "depends_on": [],
+        "display": "toggle",
+        "tooltip": "Document level security ensures identities and permissions set in Sharepoint Online are maintained in Elasticsearch. This enables you to restrict and personalize read-access users and groups have to documents in this index. Access control syncs ensure this metadata is kept up to date in your Elasticsearch documents.",
+        "default_value": null,
+        "label": "Enable document level security",
+        "sensitive": false,
+        "type": "bool",
+        "required": true,
+        "options": [],
+        "validations": [],
+        "value": false,
+        "order": 7,
+        "ui_restrictions": []
+      },
+      "fetch_users_by_site": {
+        "depends_on": [
+          {
+            "field": "use_document_level_security",
+            "value": true
```

Comment:
I'm actually already deleting this: https://github.com/elastic/connectors-python/pull/1255/files#diff-64c27d39c85cd128ec9dab21fb892a80ee2b8269470191a74a09eb0a689d39b6L999-L1007

But if you merge first, I can just pull this back out in my PR.

Reply:
Makes sense! I've merged the change and will update your PR accordingly